spdx/parsers/json_parser.py

```diff
@@ -0,0 +1,507 @@
+        for dependency in file_dependencies or []:
+            self.builder.add_file_dep(self.doc, dependency)
+        for checksum_obj in file_checksums or []:
+            if "sha1" in checksum_obj["algorithm"].lower():
+                self.builder.set_file_chksum(self.doc, checksum_obj)
+        for file_type in file_types or []:
+            self.builder.set_file_type(self.doc, file_type)
+        for artifact in file_artifacts or []:
+            for artifact_key in artifact or []:
+                self.builder.set_file_atrificat_of_project(self.doc, project_type_mapping[artifact_key],
+                                                           artifact[artifact_key])
+        for file_license in file_licenses or []:
+            self.builder.set_file_license_in_file(self.doc, document.License.from_identifier(file_license))
+
+    def parse_package(self, package):
+        self.builder.create_package(self.doc, package["name"])
+        self.builder.set_pkg_originator(self.doc, self.get_originator(package))
+        self.builder.set_pkg_down_location(self.doc, package["downloadLocation"])
+        self.builder.set_pkg_verif_code(self.doc, package["packageVerificationCode"])
+        self.builder.set_pkg_summary(self.doc, package['summary'])
+        self.builder.set_pkg_home(self.doc, package["homepage"])
+        self.builder.set_pkg_source_info(self.doc, package["sourceInfo"])
+        self.builder.set_pkg_license_comment(self.doc, package["comment"])
+        self.builder.set_pkg_vers(self.doc, package["versionInfo"])
+        self.builder.set_pkg_supplier(self.doc, self.builder.create_entity(self.doc, package["supplier"]))
+        for file in package["files"]:
+            self.parse_single_file(file["File"])
+
+    def get_sha1_chksum(self, checksums):
+        # getting only sha1 checksum because, package.py uses sha1 checksum.
+        for checksum in checksums:
+            if checksum["algorithm"].endswith("sha1"):
+                return checksum
+
+    def get_originator(self, package):
+        originator = package["originator"]
+        if originator is not None:
+            return self.builder.create_entity(self.doc, originator)
+
+
+class ExternalDocRefsParser(BaseParser):
+    def __init__(self, builder, logger):
+        super(ExternalDocRefsParser, self).__init__(builder, logger)
+
+    def parse_external_doc_refs(self, external_doc_refs):
+        for external_doc_ref in external_doc_refs:
+            self.builder.add_ext_doc_refs(self.doc, external_doc_ref["externalDocumentId"],
+                                          external_doc_ref["spdxDocumentNamespace"], external_doc_ref["checksum"])
+
+
+class Parser(AnnotationParser, ReviewParser, CreationInfoParser, LooseFieldsParser, DocDescribesParser,
+             ExternalDocRefsParser, LicenseParser):
+    def __init__(self, builder, logger):
+        super(Parser, self).__init__(builder, logger)
+
+    def parse(self, fil):
```

Comment:
Parser is a class which combines all the individual parsers to provide a single class which will manages all the parsing methods.  
All the parsers require builder and parser as argument.  
Line 483 will instantiate all the Parsers extended by the defined Parser  class.
This will provide builder and parser for all the extended parser classes.